plugins/csv_dataset.cc

```diff
@@ -595,63 +597,84 @@ struct CsvDataset::Itl: public TabularDataStore {
         if (config.quoter.length() == 1) {
             quote = config.quoter[0];
         }
-        else {
+        else if (config.quoter.length() > 1) {
             throw HttpReturnException(400, "Quoter string must have one character");
         }
+        else if (config.delimiter.length() != 0)
```

Comment:
Ideally it would be possible for the quotechar to be empty even with a delimiter.